test/schemaSelectionHandlers.test.ts

```diff
@@ -0,0 +1,99 @@
+    connection.client = {} as RemoteClient;
+    const onRequest = sandbox.fake();
+    connection.onRequest = onRequest;
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  it('add handler for "getSchema" and "getAllSchemas" requests', () => {
+    new JSONSchemaSelection(service, new SettingsState(), connection);
+    expect(connection.onRequest).calledWith(SchemaSelectionRequests.getSchema);
+    expect(connection.onRequest).calledWith(SchemaSelectionRequests.getAllSchemas);
+  });
+
+  it('getAllSchemas should return all schemas', async () => {
+    service.registerExternalSchema('https://some.com/some.json', ['foo.yaml'], undefined, 'Schema name', 'Schema description');
+    const settings = new SettingsState();
+    const testTextDocument = setupSchemaIDTextDocument('');
+    settings.documents = new TextDocumentTestManager();
+    (settings.documents as TextDocumentTestManager).set(testTextDocument);
+    const selection = new JSONSchemaSelection(service, settings, connection);
+
+    const result = await selection.getAllSchemas(testTextDocument.uri);
+
+    expect(result).length(1);
+    expect(result[0]).to.be.eqls({
+      uri: 'https://some.com/some.json',
+      fromStore: true,
+      usedForCurrentFile: false,
+      name: 'Schema name',
+      description: 'Schema description',
+    });
+  });
+
+  it('getAllSchemas should return all schemas and mark used for current file', async () => {
+    service.registerExternalSchema('https://some.com/some.json', [SCHEMA_ID], undefined, 'Schema name', 'Schema description');
+    const settings = new SettingsState();
+    const testTextDocument = setupSchemaIDTextDocument('');
+    settings.documents = new TextDocumentTestManager();
+    (settings.documents as TextDocumentTestManager).set(testTextDocument);
+    const selection = new JSONSchemaSelection(service, settings, connection);
+
+    const result = await selection.getAllSchemas(testTextDocument.uri);
+
+    expect(result).length(1);
+    expect(result[0]).to.be.eqls({
+      uri: 'https://some.com/some.json',
+      fromStore: false,
+      usedForCurrentFile: true,
+      name: 'Schema name',
+      description: 'Schema description',
+    });
+  });
+
+  it('getSchemas should return all schemas', async () => {
+    service.registerExternalSchema('https://some.com/some.json', [SCHEMA_ID], undefined, 'Schema name', 'Schema description');
+    const settings = new SettingsState();
+    const testTextDocument = setupSchemaIDTextDocument('');
+    settings.documents = new TextDocumentTestManager();
+    (settings.documents as TextDocumentTestManager).set(testTextDocument);
+    const selection = new JSONSchemaSelection(service, settings, connection);
+
+    const result = await selection.getSchemas(testTextDocument.uri);
+
+    expect(result).length(1);
+    expect(result[0]).to.be.eqls({
+      uri: 'https://some.com/some.json',
+      name: 'Schema name',
+      description: 'Schema description',
+    });
+  });
```

Comment:
Should it also have a test for the case where there are no schemas associated.

Reply:
Done